vignettes/decorate-modules-output.Rmd

```diff
@@ -0,0 +1,272 @@
+      reactive({
+        req(data())
+        data() |> within(
+          {
+            plot <- plot +
+              ggtitle("This is title") +
+              xlab(x_axis_title)
+          },
+          x_axis_title = input$x_axis_title
+        )
+      })
+    })
+  }
+)
+```
+
+As in the earlier examples, `make_teal_transform_server(expr)` can simplify the creation of the server component. 
+This wrapper allows you to use `input` object names directly in the expression:
+
+```{r}
+interactive_decorator_lang <- teal_transform_module(
+  ui = function(id) {
+    ns <- NS(id)
+    div(
+      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+    )
+  },
+  server = make_teal_transform_server(
+    expression(
+      plot <- plot +
+        ggtitle("This is title") +
+        xlab(x_axis_title)
+    )
+  )
+)
+```
+
+## Handling Various Object Names
+
+`teal_transform_module` relies on the names of objects created within a module. 
+Writing a decorator that applies to any module can be challenging because different modules may use different object names.
+It is recommended to create a library of decorator functions that can be adapted to the specific object names used in `teal` modules.
+In the following example, focus on the `output_name` parameter to see how decorator can be applied to multiple modules:
+
+    ```{r}
+gg_xlab_decorator <- function(output_name) {
+  teal_transform_module(
+    ui = function(id) {
+      ns <- NS(id)
+      div(
+        textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+      )
+    },
+    server = function(id, data) {
+      moduleServer(id, function(input, output, session) {
+        reactive({
+          req(data())
+          data() |> within(
+            {
+              output_name <- output_name +
+                xlab(x_axis_title)
+            },
+            x_axis_title = input$x_axis_title,
+            output_name = as.name(output_name)
+          )
+        })
+      })
+    }
+  )
+}
+    ```
+
+Decorator failures are managed by an internal `teal` mechanism called `trigger_on_success`, which ensures that the `data`
+object within the module remains intact. If a decorator fails, it will be ignored, and an appropriate error message will be displayed.
+
+```{r}
+failing_decorator <- teal_transform_module(
+  ui = function(id) {
+    ns <- NS(id)
+    div(
+      textInput(ns("x_axis_title"), "X axis title", value = "x axis")
+    )
+  },
+  server = function(id, data) {
+    moduleServer(id, function(input, output, session) {
+      reactive(stop("This is error"))
+    })
+  }
+)
+    ```
+
+## Example Module
+
+To include decorators in a `teal` module, pass them as arguments (`ui_args` and `server_args`) to the module’s `ui` and 
+`server` components, where they will be used by `ui/srv_teal_transform_module`:
+
+
+```{r}
+tm_decorated_plot <- function(label = "module", transforms = list(), decorators = teal_transform_module()) {
```

Comment:
It would be great if `transforms` and `decorators` had the same default/empty input type.
Is it necessary that `decorators` need to be an empty `teal_transform_module()`? 
If it could be `list()` that would be consistent with `transforms`